packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -0,0 +1,116 @@
+import path from 'path';
+import _ from 'lodash';
+import { $TSContext, pathManager, JSONUtilities } from 'amplify-cli-core';
+import { categoryName, functionParametersFileName } from './constants';
+
+export const getStoredEnvironmentVariables = (context: $TSContext, resourceName: string): object => {
+  const storedList = getStoredList(resourceName);
+  const storedReferences = getStoredReferences(resourceName);
+  const storedKeyValue = getStoredKeyValue(context, resourceName);
+  const environmentVariables = {};
+
+  // Search for key and take matched
+  // environment variables based on 4 steps
+  storedList.forEach(({ environmaneVariableName, cloudFormationParameterName }) => {
+    // 1. Check if format is correct in function-parmeter.json
+    if (!environmaneVariableName) return;
+    if (!cloudFormationParameterName) return;
+    // 2. Check if key exists in CFn template
+    if (!storedReferences[environmaneVariableName]) return;
+    if (!storedReferences[environmaneVariableName].Ref) return;
+    if (storedReferences[environmaneVariableName].Ref !== cloudFormationParameterName) return;
```

Comment:
I think the first two lines here can be dropped if you use optional chaining in the third line.

Reply:
Right, I updated. Thank you!